internal/pkg/skuba/upgrade/addon/versions_test.go

```diff
@@ -28,83 +29,126 @@ import (
-	tests := []struct {
+	type test struct {
 		name           string
+		client         clientset.Interface
 		clusterVersion *version.Version
-		clientSet      clientset.Interface
-		expected       AddonVersionInfoUpdate
+		expectedAviu   AddonVersionInfoUpdate
 		expectedErr    bool
-	}{
-		{
+	}
+
+	tests := []test{}
+	tests = append(
+		tests,
+		test{
 			name:           "no skuba-config ConfigMap",
-			clusterVersion: version.MustParseSemantic("1.15.1"),
-			clientSet:      fake.NewSimpleClientset(),
-			expected: AddonVersionInfoUpdate{
+			client:         fake.NewSimpleClientset(),
+			clusterVersion: version.MustParseSemantic("1.2.3"),
+			expectedAviu: AddonVersionInfoUpdate{
 				Current: kubernetes.AddonsVersion{},
 				Updated: kubernetes.AddonsVersion{},
 			},
 		},
-		{
-			name:           "kubernetes version 1.15.2",
-			clusterVersion: version.MustParseSemantic("1.15.2"),
-			clientSet: fake.NewSimpleClientset(&corev1.ConfigMap{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      skubaconfig.ConfigMapName,
-					Namespace: metav1.NamespaceSystem,
-				},
-				Data: map[string]string{
-					skubaconfig.SkubaConfigurationKeyName: `
-AddonsVersion:
-  cilium:
-    ManifestVersion: 32767
-    Version: 1.5.3
-  kured:
-    ManifestVersion: 32767
-    Version: 1.2.0
-  dex:
-    ManifestVersion: 32767
-    Version: 2.16.0
-  gangway:
-    ManifestVersion: 32767
-    Version: 3.1.0
-`,
-				},
-			}),
-			expected: AddonVersionInfoUpdate{
-				Current: kubernetes.AddonsVersion{
-					kubernetes.Cilium:  &kubernetes.AddonVersion{Version: "1.5.3", ManifestVersion: 32767},
-					kubernetes.Kured:   &kubernetes.AddonVersion{Version: "1.2.0", ManifestVersion: 32767},
-					kubernetes.Dex:     &kubernetes.AddonVersion{Version: "2.16.0", ManifestVersion: 32767},
-					kubernetes.Gangway: &kubernetes.AddonVersion{Version: "3.1.0", ManifestVersion: 32767},
-					kubernetes.PSP:     nil,
-				},
-				Updated: kubernetes.AddonsVersion{
-					kubernetes.PSP: &kubernetes.AddonVersion{Version: "", ManifestVersion: 1},
+		test{
+			name: "skuba-config format error",
+			client: fake.NewSimpleClientset(
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      skuba.ConfigMapName,
+						Namespace: metav1.NamespaceSystem,
+					},
+					Data: map[string]string{
+						skuba.SkubaConfigurationKeyName: `
+							AddonsVersion:
+						`,
+					},
 				},
+			),
+			clusterVersion: version.MustParseSemantic("1.2.3"),
+			expectedAviu: AddonVersionInfoUpdate{
+				Current: kubernetes.AddonsVersion{},
+				Updated: kubernetes.AddonsVersion{},
 			},
-			expectedErr: false,
+			expectedErr: true,
 		},
+	)
+
+	// Test without updated
+	for _, cv := range kubernetes.AvailableVersions() {
+		client := fake.NewSimpleClientset()
+		aviu := AddonVersionInfoUpdate{
+			Current: kubernetes.AddonsVersion{},
+			Updated: kubernetes.AddonsVersion{},
+		}
+		avs := kubernetes.AllAddonVersionsForClusterVersion(cv)
```

Comment:
nit: `avs` is kind of confusing. maybe `clusterVersions`

Reply:
I will keep it in mind, thanks for the suggestion :bowing_man: